sycl/test-e2e/Matrix/joint_matrix_bf16_fill_k_cache_runtime_dim_impl.hpp

```diff
@@ -0,0 +1,142 @@
+//------------------------------------------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-------------------------------------------------------------------------===//
+
+#include <random>
+#include <sycl/usm.hpp>
+template <size_t TM, size_t TN, size_t TK> class MatMul;
+
+template <typename T, typename TResult, size_t vnniFactor, size_t TM, size_t TN,
+          size_t TK, size_t MCache1, size_t NCache1, size_t KCache1,
+          size_t MCache2, size_t NCache2, size_t KCache2>
+void test(size_t matrix_size) {
```

Comment:
Do we have to introduce this file with mostly duplicated code?
Can we just add one more macro in the sycl/test-e2e/Matrix/joint_matrix_bf16_fill_k_cache_impl.hpp, like
```
#if runtime_matrix_size
    void test(size_t matrix_size)
#else
    void test()
    constexpr size_t matrix_size = MATRIX_SIZE;
#endif
```
and propagate the change up to `main`?

Reply:
Thanks for the suggestion! I have modified `main` to include the following, 
```
int main(
#ifdef RUNTIME_DIM
  int argc, char *argv[]
#endif //RUNTIME_DIM
  ) {

size_t matrix_size = MATRIX_SIZE;
#ifdef RUNTIME_DIM
  // Check for command line argument
  if (argc == 2) {
    matrix_size = std::stoul(argv[1]);
  } else {
    std::cerr << "Usage: ./program matrix_size\n";
    return 1; // Error if no argument
  }
#endif //RUNTIME_DIM
``` 
and modified `test` as below, 
```
void test(size_t matrix_size) 
```
This could help eliminate duplicate code for calling the `test` function in `main`. Please let me know if you’d prefer to use another way instead (having two interface for `test` function and having different calls to `test` in `main`).